operational_analysis/methods/quality_check_automation.py

```diff
@@ -0,0 +1,331 @@
+
+        for i in np.arange(24):
+            return_corr[i] = np.corrcoef(self._wtk_ws_diurnal['ws'], np.roll(self._df_diurnal,i))[0,1]
+
+        self._hour_shift = pd.DataFrame(index = np.arange(24), data = {'corr_by_hour': return_corr})
+
+    def daylight_savings_plot(self, hour_window = 3):
+          
+        """
+        Produce a timeseries plot showing daylight savings events for each year using the passed data.
+
+        Args:
+        hour_window(:obj: 'int'): number of hours outside of the Daylight Savings Time transitions to view in the plot (optional)
+
+        Returns:
+        None
+        """
+    
+        # List of daylight savings days back to 2010
+        dst_df = pd.read_csv('./daylight_savings.csv')
+
+        self._df_dst =  self._df.loc[self._df[self._id]==self._df[self._id].unique()[0], :]
+    
+        df_full = timeseries.gap_fill_data_frame(self._df_dst, self._t, self._freq) # Gap fill so spring ahead is visible
+        df_full.set_index(self._t, inplace=True)
+    
+        years = df_full.index.year.unique() # Years in data record
+        num_years = len(years)
+    
+        plt.figure(figsize = (12,20))
+
+        for y in np.arange(num_years):
+            dst_data = dst_df.loc[dst_df['year'] == years[y]] 
+        
+            # Set spring ahead window to plot
+            spring_start = pd.to_datetime(dst_data['start']) - pd.Timedelta(hours = hour_window)
+            spring_end = pd.to_datetime(dst_data['start']) + pd.Timedelta(hours = hour_window)
+        
+            # Set fall back window to plot
+            fall_start = pd.to_datetime(dst_data['end']) - pd.Timedelta(hours = hour_window)
+            fall_end = pd.to_datetime(dst_data['end']) + pd.Timedelta(hours = hour_window)
+        
+            # Get data corresponding to each
+            data_spring = df_full.loc[(df_full.index > spring_start.values[0]) & (df_full.index < spring_end.values[0])]
+            data_fall = df_full.loc[(df_full.index > fall_start.values[0]) & (df_full.index < fall_end.values[0])]
+
+            # Plot each as side-by-side subplots
+            plt.subplot(num_years, 2, 2*y + 1)
+            if np.sum(~np.isnan(data_spring[self._w])) > 0:
+                plt.plot(data_spring[self._w])
+            plt.title(years[y])
+        
+            plt.subplot(num_years, 2, 2*y + 2)
+            if np.sum(~np.isnan(data_fall[self._w])) > 0:
+                plt.plot(data_fall[self._w])
+            plt.title(years[y])
+    
+        plt.tight_layout()
+        plt.show()
+
+    def max_min(self):
+
+        """
+        Creates a DataFrame that contains the max and min values for each column
+
+        Args:
+        (None)
+
+        Returns:
+        (None)
+        """
+
+        self._max_min = pd.DataFrame(index = self._df.columns, columns = {'max', 'min'})
+        self._max_min['max'] = self._df.max()
+        self._max_min['min'] = self._df.min()
+
+    def turb_plots(self):
```

Comment:
Can we call this something more intiuitive (e.g. power_curve_plot). After all it could be revenue meter data we're plotting.

Reply:
@moptis this function specifically handles turbine plots and will fail if you try to run it for meter data. The column_plots function plots histograms for any numeric column.

I would be more than happy to change the method name; I agree that it is more intuitive.